addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -161,14 +282,82 @@ class ComposerTextInput extends Component {
+            case 'ArrowUp':
+            case 'ArrowDown':
+            case 'Tab':
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                }
+                break;
+            // ENTER: submit the message only if the dropdown mention proposition is not displayed
             case 'Enter':
-                this._onKeydownTextareaEnter(ev);
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                } else {
+                    this._onKeydownTextareaEnter(ev);
+                }
                 break;
+        }
+    }
+
+        /**
+     * @private
+     * @param ev
+     */
+    _onKeyupTextarea(ev) {
+        switch (ev.key) {
+            // ESCAPE: close mention propositions
             case 'Escape':
-                this._onKeydownTextareaEscape(ev);
+                if (this.state.isMentionsOpen) {
+                    this.state.isMentionsOpen = false;
+                    this.composer.unlink({ mentionedPartners: null });
+                    ev.stopPropagation();
+                } else {
+                    this._onKeyupTextareaEscape(ev);
+                }
                 break;
-            default:
+            // ENTER, HOME, END, UP, DOWN, PAGE UP, PAGE DOWN, TAB: check if navigation in mention propositions
+            case 'Enter':
+                if (this.state.isMentionsOpen) {
+                    this.composer._insertMentionSuggestion(this.composer.active_id, this._mentionWord);
+                    this.state.isMentionsOpen = false;
+                }
+                break;
+            case 'ArrowUp':
+            case 'PageUp':
+                if (this.state.isMentionsOpen) {
+                    this.composer._getPreviousMentionSuggestion();
+                }
+                break;
+            case 'ArrowDown':
+            case 'PageDown':
+                if (this.state.isMentionsOpen) {
+                    this.composer._getNextMentionSuggestion();
+                }
+                break;
+            case 'Home':
+                if (this.state.isMentionsOpen) {
+                    this.composer._getFirstMentionSuggestion();
+                }
                 break;
+            case 'End':
+                if (this.state.isMentionsOpen) {
+                    this.composer._getLastMentionSuggestion();
+                }
+                break;
+            case 'Tab':
+                if (this.state.isMentionsOpen) {
+                    if (ev.shiftKey) {
+                        this.composer._getPreviousMentionSuggestion();
+                    } else {
+                        this.composer._getNextMentionSuggestion();
+                    }
+                }
+                break;
+            // Otherwise, check if a mention is typed
+            default:
+                this._detectDelimiter();
         }
     }
 
```

Comment:
`_onKeydownTextareaEnter` should be after `_onKeydownTextarea`, not `_onKeyupTextarea`